src/main/resources/loot_example.yml

```diff
@@ -7,57 +7,51 @@
 # such as chance, id, amount, damage, rolls, name, level, or enchant.  They should be unique headings though as if
 # they are the same, errors may occur.
 justaname:
-  '1':
+  1: # The number could be anything, it is only used as a section title.
+     # This section will give a diamond pickaxe when this table is rolled since no chance is listed.
     id: DIAMOND_PICKAXE
-    enchant:
+    enchant: # enchant section applies to all items in its parent section - 1 in this case.
       name: silk_touch
     damage: 60
-  '2':
-    chance: 0.5
-    split: true
-    diamond:
-      id: DIAMOND_SWORD
-      enchant:
-        rolls: 2
-        split: true
+  2:
+    chance: 0.5 # This section has a 50% chance to be used when this table is rolled.
+    split: true # Section '2' will create either a power sword or a silkwood sword
+    power_sword: # A section of loot that gives diamond or iron sword. Either will be enchanted for combat.
+      split: true # Only gives a diamond OR iron sword.
+      diamond: # Just an arbitrary section name
+        id: DIAMOMD_SWORD
+      iron:
+        id: IRON_SWORD
+      enchant: # This enchant section will be applied to the diamon or iron sword when the power_sword section is rolled.
+        rolls: 2 # This enchant section will be rolled for twice.
+        split: true # Since it's split only one of the below enchant will be chosen on each of the 2 passes.
         sharpness:
           name: damage_all
-          level: -5
+          level: -5 # The level of the enchant
```

Comment:
Note that the enchant section can be shared by using a subsection like this.

Reply:
Can it? I tried running the loot table like that and it would never enchant the diamond or iron sword. It would never log that it was choosing or trying to apply and enchant either when I turned on finest-level logging. Maybe something changed in newer APIs to break it

Reply:
Hmm.. I assumed it should. Each subsection should work the same way...

Reply:
Yeah, I was just able to test it with your changes and it seems to be working fine now. I must have somehow broken it and it got fixed when you rebased it

Reply:
Awesome. Thanks for testing it!